compiler/noirc_frontend/src/tests.rs

```diff
@@ -1444,3 +1451,328 @@ fn specify_method_types_with_turbofish() {
+    let src = r#"
+    trait Default {
+        fn default() -> Self;
+    }
+
+    struct MyType<T> {
+        a: Field,
+        b: Field,
+        c: Field,
+        d: T,
+    }
+    
+    // Make sure that `T` is placed before `N` as we want to test the order is correctly maintained
+    impl<T, let N: u64> Deserialize<N> for MyType<T> where T: Default {
+        fn deserialize(fields: [Field; N]) -> Self {
+            MyType { a: fields[0], b: fields[1], c: fields[2], d: T::default() }
+        }
+    }
+    
+    trait Deserialize<let N: u64> {
+        fn deserialize(fields: [Field; N]) -> Self;
+    }
+    "#;
+    let errors = get_program_errors_elaborator(src);
+    assert!(errors.is_empty());
+}
+
+#[test]
+fn numeric_generic_used_in_where_clause() {
+    let src = r#"
+
+    trait Deserialize<let N: u32> {
+        fn deserialize(fields: [Field; N]) -> Self;
+    }
+
+    fn read<T, let N: u32>() -> T where T: Deserialize<N> {
+        let mut fields: [Field; N] = [0; N];
+        for i in 0..N {
+            fields[i] = i as Field + 1;
+        }
+        T::deserialize(fields)
+    }
+    "#;
+    let errors = get_program_errors_elaborator(src);
+    assert!(errors.is_empty());
+}
+
+#[test]
+fn implicit_numeric_generics_elaborator() {
+    let src = r#"
+    struct BoundedVec<T, MaxLen> {
+        storage: [T; MaxLen],
+        len: u64,
+    }
+    
+    impl<T, MaxLen> BoundedVec<T, MaxLen> {
+
+        // Test that we have an implicit numeric generic for "Len" as well as "MaxLen"
+        pub fn extend_from_bounded_vec<Len>(&mut self, _vec: BoundedVec<T, Len>) { 
+            // We do this to avoid an unused variable warning on `self`
+            let _ = self.len;
+            for _ in 0..Len { }
+        }
+
+        pub fn push(&mut self, elem: T) {
+            assert(self.len < MaxLen, "push out of bounds");
+            self.storage[self.len] = elem;
+            self.len += 1;
+        }
+    }
+    "#;
+    let errors = get_program_errors_elaborator(src);
+
+    for error in errors.iter() {
+        if let CompilationError::ResolverError(ResolverError::UseExplicitNumericGeneric { ident }) =
+            &errors[0].0
+        {
+            assert!(matches!(ident.0.contents.as_str(), "MaxLen" | "Len"));
+        } else {
+            panic!("Expected ResolverError::UseExplicitNumericGeneric but got {:?}", error);
+        }
+    }
+}
+
+#[test]
+fn nested_generic_elaborator() {
+    let src = r#"
+    trait Default {
+        fn default() -> Self;
+    }
+
+    struct Option<T> {
+        _is_some: bool,
+        _value: T,
+    }
+
+    impl<T> Option<T> {
+        pub fn flatten(option: Option<Option<T>>) -> Option<T> where T: Default {
+            if option._is_some {
+                option._value
+            } else {
+                Self { _is_some: false, _value: T::default() }
+            }
+        }
+    }
+    "#;
+    let errors = get_program_errors_elaborator(src);
+    assert!(errors.is_empty());
+}
```

Comment:
What is this test for? It doesn't seem to deal with numeric generics

Reply:
It does not. This was an error that I was getting in the stdlib due to having to switch the order of when generics are resolved for structs, traits, etc. I made this test for easier debugging and decided to keep it. I could provide some more comments though so that its usage is more clear.

Reply:
I ended up deleting this test to reduce this PRs diff as the logic exists in the stdlib, can be reproduced easily, and was only caused due changes while debugging.